common/types/shared.go

```diff
@@ -1362,6 +1897,41 @@ func (v *DecisionTaskTimedOutEventAttributes) GetRequestID() (o string) {
 	return
 }
 
+// Size returns the approximate memory used in bytes
+func (v *DecisionTaskTimedOutEventAttributes) Size() uint64 {
+	if v == nil {
+		return 0
+	}
+
+	size := uint64(32)
+
+	if v.TimeoutType != nil {
+		size += 8
+	}
+
+	if v.BaseRunID != "" {
+		size += uint64(16 + len(v.BaseRunID))
+	}
+
+	if v.NewRunID != "" {
+		size += uint64(16 + len(v.NewRunID))
+	}
+
+	if v.Reason != "" {
+		size += uint64(16 + len(v.Reason))
+	}
+
+	if v.Cause != nil {
+		size += 8
+	}
```

Comment:
you can have a util function which checks type of the passed argument and returns size if it's a known type.
e.g.
```
func SizeEstimate(in any) uint64 {
  switch in.(type) {
   case int64: return 8, true
    ...
    }
  return 0 
}

Reply:
this can eliminate all the hardcoded numbers spread all over. also simplifies the caller side by eliminating `!= nil` checks. you can do that check in the helper function

Reply:
I suspect the `unsafe.Sizeof()` package (credit to Ignat for pointing that out) might help some calcualtions here.